packages/core/src/indexing/local-index-api.ts

```diff
@@ -49,15 +49,21 @@ export class LocalIndexApi implements IndexApi {
    *
    * We assume that a state store always contains StreamState for an indexed stream.
    */
-  async queryIndex(query: BaseQuery & Pagination): Promise<Page<StreamState>> {
+  async queryIndex(query: BaseQuery & Pagination): Promise<Page<StreamState | null>> {
     if (this.databaseIndexApi) {
       const page = await this.databaseIndexApi.page(query)
       const edges = await Promise.all(
         // For database queries we bypass the stream cache and repository loading queue
         page.edges.map(async (edge) => {
+          let node: StreamState = await this.repository.streamState(edge.node)
+          if (node === undefined) {
+            this.logger.warn(`Cannot get stream state from repository. Unable to serve query ${JSON.stringify(query)}`)
```

Comment:
Can we make this error message more informative to the node operator?  How about:
`Did not find stream state for streamid ${edge.node.toString()} in our state store when serving an indexed query.  This may indicate a problem with data persistence of your state store, which can result in data loss.  Please check that your state store is properly configured with strong persistence guarantees.  This query may have incomplete results.  Affected query: ${JSON.stringify(query)}`